### PR TITLE
Multiple HystrixConcurrencyStrategy detected for #3256

### PR DESCRIPTION
**Pull request by ailinzhou**
Add multiple HystrixConcurrencyStrategy check. #3256

### CODE REVIEW

File: spring-cloud-netflix-hystrix/src/main/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfiguration.java [hunk 1]
```diff
@@ -56,18 +57,36 @@ public void init() {
...
-				new SecurityContextConcurrencyStrategy(existingConcurrencyStrategy));
+				new SecurityContextConcurrencyStrategy(concurrencyStrategy));
 		HystrixPlugins.getInstance().registerEventNotifier(eventNotifier);
 		HystrixPlugins.getInstance().registerMetricsPublisher(metricsPublisher);
 		HystrixPlugins.getInstance().registerPropertiesStrategy(propertiesStrategy);
 		HystrixPlugins.getInstance().registerCommandExecutionHook(commandExecutionHook);
 	}
 
+	private HystrixConcurrencyStrategy getConcurrencyStrategy() {
+		HystrixConcurrencyStrategy registeredStrategy = HystrixPlugins.getInstance()
```
Comment (ryanjbaxter) [inline spring-cloud-netflix-hystrix/src/main/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfiguration.java]: The logic here is kind of confusing the way it is written.  For example it seems that if `existingConcurrencyStrategy` is not `null` we could just return that right away.  I also dont think that the `IllegalStateException` is necessary, we should just default to the `existingConcurencyStrategy` if there is both registered (my suggestion above would eliminate the need for this).

File: spring-cloud-netflix-hystrix/src/test/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
+package org.springframework.cloud.netflix.hystrix.security;
```
Comment (ryanjbaxter) [inline spring-cloud-netflix-hystrix/src/test/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfigurationTest.java]: Missing license header.

**Comment by ryanjbaxter** [comment]
Please submit this against the 2.0.x branch

File: spring-cloud-netflix-hystrix/src/main/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfiguration.java (hunk 1 above)
Comment (ailinzhou) [reply spring-cloud-netflix-hystrix/src/main/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfiguration.java]: > For example it seems that if `existingConcurrencyStrategy` is not `null` we could just return that right away

Good idea. will `detectRegisteredConcurrencyStrategy()` be better ?



> I also dont think that the `IllegalStateException` is necessary, we should just default to the `existingConcurencyStrategy` if there is both registered

If both registered and they are not same, we really don‘t know which to select. No matter which one we choose, we lose the logic in another one. For `IllegalStateException`, this is the some style with Hystrix. (In Hystrix when register a `ConcurrencyStrategy` if there already has one, IllegalStateException was thrown)

File: spring-cloud-netflix-hystrix/src/test/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfigurationTest.java (hunk 2 above)
Comment (ailinzhou) [reply spring-cloud-netflix-hystrix/src/test/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfigurationTest.java]: All right, I will add it.

**Comment by ailinzhou** [comment]
Is this mean I should checkout a branch from 2.0.x branch commit code in it and submit a new PR to 2.0.x ? or only need to submit this commit to as a PR to 2.0.x ?

**Comment by spencergibb** [comment]
It also means rebasing your branch on 2.0.x

**Comment by ailinzhou** [comment]
Submit this PR to 2.0.x and rebase the commit to 2.0.x was done. Please see #3297

**Pull request closed without merging**